consensus/core/src/core.rs

```diff
@@ -1145,6 +1181,105 @@ mod test {
         }
     }
 
+    #[tokio::test]
+    async fn test_leader_schedule_change() {
+        telemetry_subscribers::init_for_testing();
+        let default_params = Parameters::default();
+
+        // create the cores and their signals for all the authorities
+        let mut cores = create_cores(vec![1, 1, 1, 1]);
+
+        // Now iterate over a few rounds and ensure the corresponding signals are created while network advances
+        let mut last_round_blocks = Vec::new();
```

Comment:
Can this be replaced by DagBuilder in future?

Reply:
Yeah I believe Tasos may be working on refactoring these tests to use DagBuilder, if not I will do it.